internal/resource-request-operator/resourceRequest_controller.go

```diff
@@ -0,0 +1,189 @@
+// +kubebuilder:rbac:groups=sharing.liqo.io,resources=resourceOffers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=sharing.liqo.io,resources=resourceOffers/status,verbs=get;update;patch
+
+func (r *ResourceRequestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	var resourceRequest discoveryv1alpha1.ResourceRequest
+	err := r.Get(ctx, req.NamespacedName, &resourceRequest)
+	if err != nil && !apierrors.IsNotFound(err) {
+		klog.Errorf("ResourceRequestReconciler-> unable to get resource %s: %s", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	}
+
+	if apierrors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	}
+
+	//not in delete
+	if resourceRequest.DeletionTimestamp.IsZero() {
+		// add finalizer
+		if !utils.ContainsString(resourceRequest.ObjectMeta.Finalizers, finalizerName) {
+			resourceRequest.Finalizers = append(resourceRequest.Finalizers, finalizerName)
+			if err := r.Update(ctx, &resourceRequest); err != nil {
+				klog.Errorf("%s -> unable to update resource %s: %s", resourceRequest.Spec.ClusterIdentity, resourceRequest.Spec.Namespace, err)
+				return ctrl.Result{}, err
+			}
+
+		}
+	} else {
+		//the resource is being deleted
+		if utils.ContainsString(resourceRequest.ObjectMeta.Finalizers, finalizerName) {
+			if err := r.deleteOffer(); err != nil {
+				klog.Errorf("Error deleting offer")
+				return ctrl.Result{}, err
+			}
+
+			// remove our finalizer from the list and update it.
+			resourceRequest.ObjectMeta.Finalizers = utils.RemoveString(resourceRequest.ObjectMeta.Finalizers, finalizerName)
+			if err := r.Update(ctx, &resourceRequest); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+	offerList := &sharingv1alpha1.ResourceOfferList{}
+	if err = r.getAllOffers(offerList); err != nil || len(offerList.Items) > 0 {
+		if err != nil {
+			klog.Errorf("Unable to get resource offer list")
+			return ctrl.Result{}, err
+		} else {
+			klog.Error("Resource Already exists")
+			return ctrl.Result{}, nil
+		}
+
+	}
+
+	klog.Infof("Generating Offer...")
+	offer, offerErr := r.generateResourceOffer()
+	if offerErr != nil {
+		if offer != nil {
+			klog.Errorf("unable to create resourceOffer %s", offer.ObjectMeta.Name)
+		}
+		klog.Errorf("Error: %s", offerErr)
+		return ctrl.Result{}, offerErr
+	}
+	klog.Infof("ResourceOffer successfully created")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ResourceRequestReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&discoveryv1alpha1.ResourceRequest{}).
+		Complete(r)
+}
+
+func (r *ResourceRequestReconciler) generateResourceOffer() (*sharingv1alpha1.ResourceOffer, error) {
+	offer := &sharingv1alpha1.ResourceOffer{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "liqo",
+			Name:      offerPrefix + r.ClusterId,
+		},
+		Spec: sharingv1alpha1.ResourceOfferSpec{
```

Comment:
I suggest dropping all non-mandatory fields that are not modified by your creation.